redaxo/src/core/lib/response.php

```diff
@@ -539,6 +539,26 @@ public static function sendCookie($name, $value, array $options = [])
         header($str, false);
     }
 
+    /**
+     * @param string $name    The name of the cookie
+     * @param array  $options Different cookie Options. Supported keys are:
+     *                        "expires" int|string|DateTimeInterface The time the cookie expires
+     *                        "path" string                          The path on the server in which the cookie will be available on
+     *                        "domain" string|null                   The domain that the cookie is available to
+     *                        "secure" bool                          Whether the cookie should only be transmitted over a secure HTTPS connection from the client
+     *                        "httponly" bool                        Whether the cookie will be made accessible only through the HTTP protocol
+     *                        "samesite" string|null                 Whether the cookie will be available for cross-site requests
+     *                        "raw" bool                             Whether the cookie value should be sent with no url encoding
+     * @psalm-param array{expires?: int|string|DateTimeInterface, path?: string, domain?: ?string, secure?: bool, httponly?: bool, samesite?: ?string, raw?: bool} $options
+     *
+     * @throws \InvalidArgumentException
+     */
+    public static function clearCookie(string $name, array $options = []): void
+    {
+        $options['expires'] = 1;
+        self::sendCookie($name, null, $options);
```

Comment:
Ich denke der aufrufer sollte cookie selbst neu aufrufen.. clearCookie() impliziert für mich nicht dass ein neuer gesendet wird

Reply:
Der Weg hierhin war so:
- Hmm, wie löscht man denn nochmal Cookies?
- Ich schau mal [bei Symfony nach](https://github.com/symfony/symfony/blob/af160e2c3f8c390583142764d2c73714c6a19a7f/src/Symfony/Component/HttpFoundation/ResponseHeaderBag.php#L242)
- Ah, expires in der Vergangenheit
- Um beim nächsten Mal nicht wieder nachschauen zu müssen, integrier ich die clearCookie-Methode bei uns auch mal

Reply:
Ok ggf ist es nur verwirrend dass man ein options array als parameter hat?

Reply:
> Würde man nicht erwarten dass man nur den namen angeben muss

Eigentlich ja. Aber wie ist das, kann es mehrere gleichnamige geben, mit unterschiedlichen Options, z.B. unterschiedlichem Pfad?

Reply:
Unterschiedliche domain/pfad könnte sein